src/funjson.c

```diff
@@ -667,13 +670,9 @@ FUNCTION(fun_json)
     jstr = cJSON_PrintUnformatted(obj);
     cJSON_Delete(obj);
     latin1 = utf8_to_latin1(jstr, -1, &len, 0, "json.string");
-    for (c = latin1; *c; c += 1) {
-      if (!isprint(*c) && !isspace(*c)) {
-        *c = '?';
-      }
-    }
-
-    safe_strl(latin1, len, buff, bp);
+    partialbuf = json_unescape_latin1string(latin1,len);
+    safe_strl(partialbuf, strlen(partialbuf), buff, bp);
```

Comment:
As above with safe_str vs safe_strl

Reply:
Moving to safe_str somehow is adding a character, which suggests the \0 is being improperly placed.